lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py

```diff
@@ -287,44 +334,41 @@ def main():
         nics_service = host_service.nics_service()
         nic = search_by_name(nics_service, nic_name)
 
-        network_names = [network['name'] for network in networks]
+        network_names = [network['name'] for network in networks or []]
         state = module.params['state']
         if (
             state == 'present' and
             (nic is None or host_networks_module.has_update(nics_service.service(nic.id)))
         ):
+            removed_bonds = []
             # Remove networks which are attached to different interface then user want:
             attachments_service = host_service.network_attachments_service()
-            remove_network_attachments = [
-                a for a in attachments_service.list()
-                if get_link_name(connection, a.network) in network_names and
-                get_link_name(connection, a.host_nic) != interface
-            ]
-            if remove_network_attachments:
-                host_networks_module.action(
-                    entity=host,
-                    action='setup_networks',
-                    post_action=host_networks_module._action_save_configuration,
-                    check_connectivity=module.params['check'],
-                    removed_network_attachments=remove_network_attachments,
-                )
+
+            # Append attachment ID to network if needs update:
+            for a in attachments_service.list():
+                current_network_name = get_link_name(connection, a.network)
+                if current_network_name in network_names:
+                    for n in networks:
+                        if n['name'] == current_network_name:
+                            n['id'] = a.id
+
+            # Check if we have to break some bonds:
+            for host_nic in nics_service.list():
+                if host_nic.bonding and nic.id in [slave.id for slave in host_nic.bonding.slaves]:
```

Comment:
It could be possible, that `nic` is `None` and in this case, this check will fail

~~~
  File "/tmp/ansible_WEMKC_/ansible_module_ovirt_host_networks.py", line 311, in main
    if host_nic.bonding and nic.id in [slave.id for slave in host_nic.bonding.slaves]:
AttributeError: 'NoneType' object has no attribute 'id'
~~~

Reply:
Right thanks, in case there exist different bond and we are trying to create another which don't yet exist it may fail. Fixed.